src/query/storage/m3/storage.go

```diff
@@ -160,6 +160,15 @@ func (s *m3storage) FetchRaw(
 
 	wg.Wait()
 
+	// Check if the query was interrupted.
+	select {
+	case <-ctx.Done():
+		return nil, noop, ctx.Err()
+	case <-options.KillChan:
```

Comment:
Do we actually need a separate `killChan` in addition to the context? Could a cancellable context be used to send that signal all the way through?